File: src/main/java/ru/citeck/ecos/model/doclib/service/DocLibChildrenQuery.java
package ru.citeck.ecos.model.doclib.service;

import lombok.Data;
import org.jetbrains.annotations.Nullable;
import ru.citeck.ecos.records2.RecordRef;
import ru.citeck.ecos.records2.predicate.model.Predicate;

@Data
public class DocLibChildrenQuery {

    @Nullable
    private RecordRef parentRef;
    @Nullable
    private Predicate filter;
    @Nullable
    private DocLibNodeType nodeType;

    private boolean recursive;
}

File: src/main/java/ru/citeck/ecos/model/domain/doclib/DocLibChildrenQuery.kt
package ru.citeck.ecos.model.domain.doclib

import ru.citeck.ecos.records2.predicate.model.Predicate
import ru.citeck.ecos.webapp.api.entity.EntityRef

class DocLibChildrenQuery(
    val parentRef: EntityRef? = null,
    val filter: Predicate,
    val nodeType: DocLibNodeType? = null
)
